frontend/src/routes/Infrastructure/Clusters/ManagedClusters/ImportCluster/ImportCluster.tsx

```diff
@@ -314,6 +321,22 @@ export default function ImportClusterPage() {
           return state.importMode === ImportMode.discoveryOCM ? { ...state, credential: action.credential } : state
         case 'setKubeconfig':
           return state.importMode === ImportMode.kubeconfig ? { ...state, kubeconfig: action.kubeconfig } : state
+        case 'updateCredentials': {
+          const updatedCredentialList = action.payload?.credentialList || []
+          const namespaceExists = updatedCredentialList.some(
+            (credential) => credential.metadata.namespace === state.namespace
+          )
+          const credentialExists = updatedCredentialList.some(
+            (credential) => credential.metadata.name === state.credential
+          )
+
+          return {
+            ...state,
+            isNamespaceAvailable: namespaceExists,
+            isCredentialAvailable: credentialExists,
+            credentialList: updatedCredentialList,
+          }
```

Comment:
What I wanted you to do was think about these situations:
- Is the currently selected namespace (if any) still available
- Is the currently selected credential (if any) still available
And then decide the best way to handle them. 

For example, suppose in the import form I have selected namespace `default` and credential `ocm-cred`, and that is the only OCM credential in that namespace. Someone else deletes the credential. There is one other OCM credential defined in namespace `my-project`. Do we update the state so that `namespace` and `credential` are empty string, clearing out the dropdowns in the form? Or do we automatically select the first namespace and the first credential in that namespace?

So some of the code you have here is a good start - you will want to check if we have a namespace selected if it still exists. For the credential, note that the name is only unique within a namespace, so we could have credential `ocm-creds` defined in two different namespaces. Might be a bit confusing if which namespace we were using changed on us.

Reply:
thank you for the feedback @KevinFCormier, I will work on fixing it